.github/workflows/install-macos.yml

```diff
@@ -0,0 +1,29 @@
+name: Install on macOS
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  install:
+    runs-on: macos-latest
+
+    strategy:
+      matrix:
+        shell:
+          - bash
+          - dash
+          - ksh
+          - sh
+          - zsh
+        script:
+          - install.sh
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install shell
+        run: brew install ${{ matrix.shell }}
+        if: ${{ matrix.shell == 'ksh' || matrix.shell == 'dash' }}
+      - name: Run installation script
+        run: SHELL="$(which ${{ matrix.shell }})" ${{ matrix.shell }} ${{ matrix.script }}
```

Comment:
Why should the SHELL env variable be set?

Reply:
I recognize that `pnpm setup` refers SHELL env to configure the shell profile.
I want to check in GitHub Actions whether this works.
Running `${{ matrix.shell }} ${{ matrix.script }}` will leave the SHELL env the system's default.
I tried `-l` option but the SHELL env won't change.

Reply:
Hm, maybe we should use a different way to recognize the active SHELL. Or maybe it is OK to use the default SHELL.

Reply:
I think CI needs SHELL to be changed for a testing purpose but a user doesn't need to be changed.
As a user, `pnpm setup` should recognize the login shell, not the running shell.
The login shell and the script running shell often differ.